### PR TITLE
Quantization levels and dithering options

### LINKED ISSUE #304: Add support for lossless compression of floating point images
Currently, `fitsio` does not have support for lossless compression of floating-point images.  I believe this can be turned on with the `fits_set_quantize_level(fits, 0, status))` call in `cfitsio`.

At the same time, it seems that `fitsio` is broken when reading these sorts of compressed images, as illustrated here (because `astropy.io.fits` does support writing in this mode):

```
import astropy.io.fits as fits
import numpy as np
import healpy as hp
import fitsio
cov_index_map = np.arange(1000, dtype=np.int64)
sparse_map_int = np.arange(10000, dtype=np.int32)
sparse_map_int[500: 1000] = np.iinfo(np.int32).min
sparse_map_float = np.arange(10000, dtype=np.float32)
sparse_map_float[500: 1000] = hp.UNSEEN
hdu_list = fits.HDUList()
hdu = fits.PrimaryHDU(data=cov_index_map, header=fits.Header())
hdu_list.append(hdu)
hdu = fits.CompImageHDU(data=sparse_map_float, header=fits.Header(),
                        compression_type='GZIP_2', tile_size=(1000, ),
                        quantize_level=0.0)
hdu_list.append(hdu)
hdu_list.writeto('test.fits', overwrite=True)
sparse_map_in = fits.getdata('test.fits', ext=1)
np.testing.assert_array_almost_equal(sparse_map_in, sparse_map_float)
sparse_map_in = fitsio.read('test.fits', ext=1)
np.testing.assert_array_almost_equal(sparse_map_in, sparse_map_float)
```

(This fails on the final assertion).

### PR DESCRIPTION
**Pull request by esheldon**
closes #304 

TODO:

- [x] add dithering method configuration

Possible TODO

- [x] Add HCOMPRESS configuration keywords

### CODE REVIEW

**Comment by esheldon** [comment]
@erykoff

File: fitsio/fitslib.py [hunk 1]
```diff
@@ -345,6 +347,11 @@ def write(filename, data, extname=None, extver=None, header=None,
         (case-insensitive) See the cfitsio manual for details.
     tile_dims: tuple of ints, optional
         The size of the tiles used to compress images.
+    qlevel: float, optional
+        Quantization level.  Lower generally result in more compression, we
+        recommend one reads the FITS standard or cfitsio manual to fully
+        understand the effects of quantization.  None or 0 means no
+        quantization, and for gzip also implies lossless.  Default is 4.0
```
Comment (erykoff) [inline fitsio/fitslib.py:354]: I think the `qlevel` documentation (which unfortunately must be repeated a lot) should mention that this is only relevant for floating-point data.

**Comment by esheldon** [comment]
@erykoff ready for testing

**Comment by erykoff** [comment]
Works for me, tests run fine, and I was able to write unquantized gzip-compressed floating-point data and read it correctly with both `fitsio` and `astropy`.

**Pull request merged**